internal/poller/gotimestatus/gotimestatus.go

```diff
@@ -0,0 +1,188 @@
+	ID        string     `json:"id"`
+	URL       string     `json:"url"`
+	CreatedAt createTime `json:"created_at"`
+}
+
+type createTime struct {
+	time.Time
+}
+
+func (ct *createTime) UnmarshalJSON(bs []byte) error {
+	var str string
+	if err := json.Unmarshal(bs, &str); err != nil {
+		return err
+	}
+	t, err := time.Parse("2006-01-02T15:04:05.999Z", str)
+	if err != nil {
+		return nil
+	}
+	ct.Time = t
+	return nil
+}
+
+const (
+	// @gotime@changelog.social
+	// acquired from curl 'https://changelog.social/api/v1/accounts/lookup?acct=gotime'
+	// It's immutable so we don't need to look it up again
+	gotimeAccountID = "109349735213354404"
+
+	gotimeStatusesAPI = "https://changelog.social/api/v1/accounts/" + gotimeAccountID + "/statuses"
+)
+
+// Poll conditionally calls notify if there is a status update from @gotime@changelog.social
+//
+// For a status to be posted, it needs to be younger than the maxStatusAge
+// this prevents very old statuses from being notified and acts as a safeguard
+// if the last status ID could not be persisted to state storage, and prevents reposts if we lose the last status ID.
+func (gt *GoTimeStatus) Poll(ctx context.Context) error {
+	gt.logger.Trace().Msg("gotime status poll")
+	now := timeNow()
+	accountStatusURL := gotimeStatusesAPI
+	if gt.lastStatus == "" {
+		gt.logger.Trace().Msg("getting latest status")
+	} else {
+		gt.logger.Trace().Msgf("getting statuses since %s", gt.lastStatus)
+		accountStatusURL = fmt.Sprintf("%s?since_id=%s", gotimeStatusesAPI, url.QueryEscape(gt.lastStatus))
+	}
+	var statuses []mastodonStatus
+	err := gt.get(ctx, accountStatusURL, &statuses)
+	if err != nil {
+		return err
+	}
+	if len(statuses) == 0 {
+		// No new statuses
+		gt.logger.Trace().Msg("no statuses found")
+		return nil
+	}
+	if gt.lastStatus == "" {
+		// no last status, only notify on the latest status
+		// which should be the first element in the list
+		statuses = statuses[0:1]
+	} else {
+		// Sort statuses by age in ascending order
+		sort.Slice(statuses, func(i, j int) bool {
+			return statuses[i].CreatedAt.Time.Before(statuses[j].CreatedAt.Time)
+		})
+	}
+
+	for _, status := range statuses {
+		gt.lastStatus = status.ID
+		age := now.Sub(status.CreatedAt.Time)
+		if age > gt.statusMaxAge { // too old
+			gt.logger.Trace().Msgf("status %s skipped. too old: %s", status.ID, age)
+			continue
+		}
+		gt.logger.Trace().Msgf("notify gotime statuses: %s", status.URL)
+		if err := gt.notify(ctx, status.URL); err != nil {
+			return fmt.Errorf("failed to notify social status %s: %w", status.URL, err)
+		}
+	}
+	if err := gt.store.Put(ctx, gt.lastStatus); err != nil {
+		return fmt.Errorf("failed to persist status ID to redis: %w", err)
+	}
+	return nil
+}
+
+// get makes an HTTP request to url and unmarshals the JSON response into i.
+func (gt *GoTimeStatus) get(ctx context.Context, url string, i interface{}) error {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		panic(err)
```

Comment:
I'd rather `return err` than panic here.

Reply:
Can do. I copied this from `internal/poller/gotime` so its pulling in some of that here. Should I also update that package with these recomendations?